Aztec/Classes/Public/Formatters/AttributeFormatter.swift

```diff
@@ -0,0 +1,148 @@
+import UIKit
+
+/// A type that provides support for toggling compound attributes in an
+/// attributed string.
+///
+/// When you want to represent an attribute that does not have a 1-1
+/// correspondence with a standard attribute, it is useful to have a virtual
+/// attribute. Toggling this attribute would also toggle the attributes for its
+/// defined style.
+///
+protocol AttributeFormatter {
```

Comment:
I think it would be great to have unit tests for the default method implementations in this file.

If you'd like to push things forwards I'm perfectly fine with handling that in a separate PR though.  Especially since there's some indirect testing through `BlockquteFormatterTests`.